src/main/java/software/amazon/awssdk/crt/auth/credentials/Credentials.java

```diff
@@ -14,22 +14,31 @@ public class Credentials {
     private byte[] secretAccessKey;
     private byte[] sessionToken;
 
-    /**
-     * Default constructor
-     */
-    public Credentials() {}
+    private Credentials() {}
```

Comment:
are you sure SDK is not using this currently and making this private will not break build for them?

Reply:
I have tried searching it in their code and didn't find any usage. It doesn't break any of their test either.

Reply:
Both solutions seem fine to me. As it wouldn't have worked before anyhow, it will just replace a runtime error with compile time if someone was using it. @graebm What do you think?

Reply:
Changing it to private is an API-breaking change. Don't do that. Like Dmitriy said, we're a public library and don't know what ALL customers' code looks like.

Keep it public. But in the documentation, encourage people to use `createAnonymousCredentials()` instead since it's more clear